terminal/TerminalClient.cpp

```diff
@@ -130,7 +137,139 @@ void handleWindowChanged(winsize* win) {
   }
 }
 
-int main(int argc, char** argv) {
+void initSetupSSH() {
+  string CLIENT_TERM(getenv("TERM"));
+  if (FLAGS_passkey.empty()) {
+    FILE *passkey_p = popen(
+        "env LC_ALL=C tr -dc \"a-zA-Z0-9\" < /dev/urandom | head -c 32", "r");
+    if (!passkey_p) {
+      cout << "cannot generate passkey!" << endl;
+      exit(1);
+    }
+    char passkey_buffer[1024];
+    fgets(passkey_buffer, sizeof(passkey_buffer), passkey_p);
+    pclose(passkey_p);
+    FLAGS_passkey = string(passkey_buffer);
+  }
+  if (FLAGS_id.empty()) {
+    FILE *id_p = popen(
+        "env LC_ALL=C tr -dc \"a-zA-Z0-9\" < /dev/urandom | head -c 16", "r");
+    if (!id_p) {
+      cout << "cannot generate id!" << endl;
+      exit(1);
+    }
+    char id_buffer[1024];
+    fgets(id_buffer, sizeof(id_buffer), id_p);
+    pclose(id_p);
+    FLAGS_id = string(id_buffer);
+  }
+
+
+  string cmdoptions = "--idpasskey=" + FLAGS_id + "/" + FLAGS_passkey;
```

Comment:
idpasskey is not secure because command line options are visible to all users (ps aux).  The original et script put the passkey & id in a file and used the idpasskeyfile option which reads from that file and then destroys it.